xarray/core/pdcompat.py

```diff
@@ -73,6 +75,19 @@ def __repr__(self) -> str:
 NoDefault = Literal[_NoDefault.no_default]  # For typing following pandas
 
 
+def timestamp_as_unit(date: pd.Timestamp, unit: str) -> pd.Timestamp:
+    # compatibility function for pandas issue
+    # where "as_unit" is not defined for pandas.Timestamp
+    # in pandas versions < 2.2
+    # can be removed minimum pandas version is >= 2.2
+    unit = cast(PDDatetimeUnitOptions, unit)
```

Comment:
Is there a reason you don't use this type directly in the function annotation?

Reply:
I think I tried that, but that broke the typing in the calling functions code which I could not get to work. I'll retry.

Reply:
Works now!